PowerShellEditorServices.build.ps1

```diff
@@ -11,11 +11,21 @@ param(
 
     [string]$DefaultModuleRepository = "PSGallery",
 
-    [string]$TestFilter = ''
+    # See: https://docs.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests
+    [string]$TestFilter = '',
+
+    # See: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-test
+    # E.g. use @("--logger", "console;verbosity=detailed") for detailed console output instead
+    [string[]]$TestArgs = @("--logger", "trx")
 )
 
 #Requires -Modules @{ModuleName="InvokeBuild";ModuleVersion="3.2.1"}
 
+$script:dotnetTestArgs = @("test")
+$script:dotnetTestArgs += $TestArgs
+$script:dotnetTestArgs += if ($TestFilter) { @("--filter", $TestFilter) } else { "" }
+$script:dotnetTestArgs += "--framework"
```

Comment:
The way I usually do this is like this:

```powershell
$script:dotnetTestArgs = @(
    "test"
    $TestArgs
    if ($TestFilter) { "--filter", $TestFilter }
    "--framework"
)
```

The benefits are:
- No array reallocation
- No need for an awkward `else` clause
- No repetition of the LHS
- Purely declarative statement of the array

Reply:
Much better, thanks!